drivers/sdhc/sdhci_common.c

```diff
@@ -428,6 +428,72 @@ static int sdhci_transfer_data(struct sdhci_common *sdhci_ctx, struct sdhc_data
 	return sdhci_poll_data_complete(sdhci_ctx, timeout / 100);
 }
 
+static void sdhc_adma_desc_fill(struct sdhc_adma_desc *desc, uintptr_t dma_addr, uint16_t len,
+				bool end)
+{
+	uint16_t cmd;
+
+	cmd = SDHC_ADMA_DESC_VALID | SDHC_ADMA_DESC_ACT_TRAN;
+	if (end) {
+		cmd |= SDHC_ADMA_DESC_END;
+	}
+
+	desc->cmd = cmd;
```

Comment:
As far as I can see, this pertains to a hardware structure, and you're populating a structure in a shared file that needs to be usable across different devices. Should you fix descriptor filling for both little-endian and big-endian machines?